pkg/sriov/sriov.go

```diff
@@ -22,21 +22,34 @@ import (
 	"io/ioutil"
 	"os"
 	"path/filepath"
+	"regexp"
 
-	"github.com/Mellanox/sriovnet"
 	"github.com/containernetworking/cni/pkg/skel"
 	current "github.com/containernetworking/cni/pkg/types/100"
 	"github.com/containernetworking/plugins/pkg/ip"
 	"github.com/containernetworking/plugins/pkg/ns"
+	"github.com/k8snetworkplumbingwg/sriovnet"
 	"github.com/vishvananda/netlink"
 )
 
 var (
 	// SysBusPci is sysfs pci device directory
-	SysBusPci = "/sys/bus/pci/devices"
+	SysBusPci       = "/sys/bus/pci/devices"
+	rePciDeviceName = regexp.MustCompile(`^[0-9a-f]{4}:[0-9a-f]{2}:[01][0-9a-f]\.[0-7]$`)
+	reAuxDeviceName = regexp.MustCompile(`^\w+.\w+.\d+$`)
 )
 
-// GetVFLinkName retrives interface name for given pci address
+// IsPCIDeviceName check if passed device id is a PCI device name
+func IsPCIDeviceName(deviceID string) bool {
+	return rePciDeviceName.MatchString(deviceID)
+}
+
+// IsAuxDeviceName check if passed device id is a Auxiliary device name
+func IsAuxDeviceName(deviceID string) bool {
```

Comment:
it might be safer to check if given device is found on the auxiliary bus like in [1]
the regex is quite loose for SFs

[1] https://github.com/containernetworking/plugins/pull/1003/files